pkg/volume/util/operationexecutor/operation_generator.go

```diff
@@ -1681,3 +1691,56 @@ func isDeviceOpened(deviceToDetach AttachedVolume, hostUtil hostutil.HostUtils)
 	}
 	return deviceOpened, nil
 }
+
+func cancelControlPlaneExpansion(pvc *v1.PersistentVolumeClaim, pv *v1.PersistentVolume, client clientset.Interface) (bool, error) {
+	if !utilfeature.DefaultFeatureGate.Enabled(features.RecoverVolumeExpansionFailure) {
+		return false, nil
+	}
+
+	newSize := pvc.Spec.Resources.Requests[v1.ResourceStorage]
+	statusSize := pvc.Status.Capacity[v1.ResourceStorage]
+	pvSize := pv.Spec.Capacity[v1.ResourceStorage]
+	if statusSize.Cmp(newSize) >= 0 && pvSize.Cmp(newSize) >= 0 {
```

Comment:
If newSize > statusSize, it will remain this way?  So after canceling is complete, we'll still see the requested size of PVC is greater than the actual PVC size shown by "kubectl describe pvc"?  The pvSize and statusSize will be the same at the end, right?  Do we need to compare the pvSize and statusSize here?  It seems that there could be different combinations of newSize, statusSize, and pvSize here.  It's good to add comments to explain the possible combinations and expected outcome.

Reply:
sure, I will add a comment but if `newSize > statusSize` then volume expansion *should not* be cancelled and it should retried/proceed as before.